Remove port when using mongodb+srv

jareeve opened the pull request:
### Description
When using protocol mongodb+srv the port must not be set in the connection url.

#### Related issues
https://github.com/strongloop/loopback-connector-mongodb/issues/496

### Checklist

<!--
- Please mark your choice with an "x" (i.e. [x], see
https://github.com/blog/1375-task-lists-in-gfm-issues-pulls-comments)
- PR's without test coverage will be closed.
-->

- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

slnode commented:
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

dhmlau commented:
@jareeve, thanks for the PR.  Could you please sign the CLA https://cla.strongloop.com/agreements/strongloop/loopback-connector-mongodb?  Thanks.

dhmlau commented on lib/mongodb.js:
```diff
@@ -57,6 +57,11 @@ function generateMongoDBURL(options) {
   options.port = options.port || 27017;
   options.database = options.database || options.db || 'test';
   var username = options.username || options.user;
+  var portUrl = ''
```
missing the ending `;`

jareeve replied:
added the colon and linter is now passing

jareeve commented:
Looks like travis is failing with:
----------------------------------------------------
  1 failing
  1) mongodb connector
       does not execute a nested `$where`:
     Uncaught AssertionError: expected Error {
  code: 'OPERATOR_NOT_ALLOWED_IN_QUERY',
  statusCode: 400,
  details: Object {
    operators: Array [ '$where' ],
    where: Object {
      content: Object {
        $where: 'function() {return this.content.contains("content")}'
      }
    }
  },
  message: 'Operators "$where" are not allowed in query'
} to not exist
      at Post.find (test/mongodb.test.js:830:24)
      at /home/travis/build/strongloop/loopback-connector-mongodb/node_modules/loopback-datasource-juggler/lib/dao.js:1525:7
      at process._tickCallback (internal/process/next_tick.js:61:11)
---------------------------------------------------------------------------------------

But I saw this failure before I made any code changes. Is it a already know failure?

dhmlau commented:
@jannyHou, seems like it might be related to the PR that you've worked on: https://github.com/strongloop/loopback-connector-mongodb/pull/484.  Could you please help on the test failure? Thanks.

jannyHou commented:
@dhmlau My fix doesn't seem to be merged into master. If master fails then we should fix it first.

jannyHou commented on lib/mongodb.js:
```diff
@@ -57,6 +57,11 @@ function generateMongoDBURL(options) {
   var username = options.username || options.user;
+  var portUrl = '';
+  // only include port if not using mongodb+srv
+  if(options.protocol !== 'mongodb+srv') {
```
Nitpick: can we use a truthy assertion here?
Like 

```js
if(options.protocal === 'mongodb+srv') {
  options.port = '';
}
```

And then you don't have to make the code change on line 74 and 83.

jareeve replied:
Not sure I follow, I need the change on 74 and 83 to remove the : as well as the port number. Pretty sure the url will not be valid with : in it.

jareeve commented:
@jannyHou I have added a unit test and added some details to the readme about using mongo+srv. I have posted a comment above about your proposed change as i think it will not work.

jareeve commented:
@jannyHou let me know if there is anything else you need me to do on this pr.

dhmlau commented:
@slnode test please

dhmlau commented:
@slnode test please
There's PR merged 12 days ago has most of the CI passed.  Want to kick it off again for this PR. 

@jannyHou, please review again. Thanks

dhmlau commented:
CI failed because of the error [mentioned above](https://github.com/strongloop/loopback-connector-mongodb/pull/497#issuecomment-461376513).

emonddr commented on test/mongodb.test.js:
```diff
@@ -3073,6 +3073,60 @@ describe('mongodb connector', function() {
+        };
+        module.generateMongoDBURL(options).should.be.eql('mongodb://fakeHostname:9999/fakeDatabase');
+      });
+      it('when protocol is mongodb and no username/password', function () {
```
name of test should be 'when protocol is mongodb and username/password' since this test
IS passing user name and password in the options

jareeve replied:
Corrected to state it has username and password

emonddr commented on test/mongodb.test.js:
```diff
@@ -3073,6 +3073,60 @@ describe('mongodb connector', function() {
+        // mongodb+srv url should not have the port in it
+        module.generateMongoDBURL(options).should.be.eql('mongodb+srv://fakeHostname/fakeDatabase');
+      });
+      it('when protocol is mongodb+srv and no username/password', function () {
```
Name of test should be 'when protocol is mongodb+srv and username/password' since the
user name and password IS being passed in the options.

jareeve replied:
Corrected to state it has username and password

dhmlau commented:
@jareeve , could you please address the review comment from @emonddr ? Thanks.

jareeve commented:
@emonddr @dhmlau sorry for the delay but i missed the original comments. i have now updated the test names.

emonddr commented:
Please fix the lint problem with the long commit message
Click `Details`
![image](https://user-images.githubusercontent.com/6864736/55557711-76e1ee00-56b8-11e9-8f35-fe1f3d582cb8.png)
and you will see:
![image](https://user-images.githubusercontent.com/6864736/55557686-6762a500-56b8-11e9-8d13-79c4f1bbeb47.png)

emonddr commented:
@jannyHou 

![image](https://user-images.githubusercontent.com/6864736/55557960-11dac800-56b9-11e9-8f36-b256039b461e.png)

jareeve commented:
I think I have fixed everything due to me.

dhmlau commented:
@slnode test please

dhmlau commented:
@jareeve, thanks! one last thing, could you please squash your commits? 
See instructions https://loopback.io/doc/en/lb4/submitting_a_pr.html#10-final-rebase-and-squashing-of-commits. Thanks.

dhmlau commented:
@emonddr , could you please help land this PR? Thanks!

dhmlau commented:
@jannyHou @emonddr, I've rebased the PR and fixed the commit message linting error. Could you please review again? Thanks.

jannyHou commented:
@slnode test please

dhmlau commented:
@jareeve  thanks for your contribution.  Your PR has landed! 🎉
